web/test/pages/profile-owning.test.tsx

```diff
@@ -0,0 +1,679 @@
+/* eslint-disable jest/expect-expect */
+import { getMergedConfig } from "@/contexts/configContext";
+import * as api from "@/lib/api-client/apiClient";
+import { IsAuthenticated } from "@/lib/auth/AuthContext";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import analytics from "@/lib/utils/analytics";
+import { templateEval } from "@/lib/utils/helpers";
+import Profile from "@/pages/profile";
+import { randomPublicFilingLegalStructure, randomTradeNameLegalStructure } from "@/test/factories";
+import { withAuthAlert } from "@/test/helpers/helpers-renderers";
+import * as mockRouter from "@/test/mock/mockRouter";
+import { useMockRouter } from "@/test/mock/mockRouter";
+import { useMockRoadmap } from "@/test/mock/mockUseRoadmap";
+import { WithStatefulProfileFormContext } from "@/test/mock/withStatefulProfileData";
+import {
+  currentUserData,
+  setupStatefulUserDataContext,
+  WithStatefulUserData,
+} from "@/test/mock/withStatefulUserData";
+import {
+  defaultDateFormat,
+  emptyProfileData,
+  generateMunicipality,
+  generateProfileData,
+  getCurrentDate,
+  LookupOwnershipTypeById,
+  LookupSectorTypeById,
+  Municipality,
+  randomInt,
+  UserData,
+} from "@businessnjgovnavigator/shared";
+import {
+  generateTaxFilingData,
+  generateUserData as _generateUserData,
+  randomLegalStructure,
+} from "@businessnjgovnavigator/shared/test";
+
+import {
+  chooseRadio,
+  chooseTab,
+  clickBack,
+  clickSave,
+  fillText,
+  getBusinessNameValue,
+  getBusinessProfileInputFieldName,
+  getDateOfFormation,
+  getEmployerIdValue,
+  getEntityIdValue,
+  getMunicipalityValue,
+  getNotesValue,
+  getTaxIdValue,
+  selectByText,
+  selectByValue,
+} from "@/test/pages/profile-helpers";
+import { createTheme, ThemeProvider } from "@mui/material";
+import { fireEvent, render, screen, waitFor, within } from "@testing-library/react";
+
+const date = getCurrentDate().subtract(1, "month").date(1);
+const Config = getMergedConfig();
+
+const dateOfFormation = date.format(defaultDateFormat);
+const mockApi = api as jest.Mocked<typeof api>;
+
+function setupMockAnalytics(): typeof analytics {
+  return {
+    ...jest.requireActual("@/lib/utils/analytics").default,
+    event: {
+      ...jest.requireActual("@/lib/utils/analytics").default.event,
+      profile_location_question: {
+        submit: {
+          location_entered_for_first_time: jest.fn(),
+        },
+      },
+    },
+  };
+}
```

Comment:
I think we use this in most of the files, too. Might be another good add to the helpers.

Reply:
Jest didn't like this being imported from a different file because I think it uses jest within it, so left this in the respective files.